pkg/controller/grouppermission/grouppermission_controller.go

```diff
@@ -100,54 +86,196 @@ func (r *ReconcileGroupPermission) Reconcile(request reconcile.Request) (reconci
+	// checkClusterRoleName is a helper func, returns bool
+	if checkClusterRoleName(instance, clusterRoleList) {
+		return reconcile.Result{}, nil
+	} else {
+		// get the conditions array
+		groupPermissionConditions := instance.Status.Conditions
+
+		// make new condition
+		newCondition := managedv1alpha1.Condition{
+			LastTransitionTime: metav1.Now(),
+			Message:            "ClusterRole for clusterPermission does not exist",
+			ClusterRoleName:    "clusterRoleName", //this needs to get fixed
+			Status:             true,
+			State:              "Failed",
+		}
+
+		// append new condition to conditions array
+		groupPermissionConditions = append(groupPermissionConditions, newCondition)
+
+		// update GroupPermission object with new conditions
+		instance.Status.Conditions = groupPermissionConditions
+
+		err = r.client.Status().Update(context.TODO(), instance)
 		if err != nil {
-			return reconcile.Result{}, err
+			reqLogger.Error(err, "Failed to update condition.")
+			return reconcile.Result{}, nil
 		}
+	}
 
-		// Pod created successfully - don't requeue
+	clusterRoleBindingList := &v1.ClusterRoleBindingList{}
+	err = r.client.Get(context.TODO(), request.NamespacedName, clusterRoleBindingList)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
 		return reconcile.Result{}, nil
-	} else if err != nil {
-		return reconcile.Result{}, err
 	}
 
-	// Pod already exists - don't requeue
-	reqLogger.Info("Skip reconcile: Pod already exists", "Pod.Namespace", found.Namespace, "Pod.Name", found.Name)
+	// get permission type
+	permission := &managedv1alpha1.Permission{}
+
+	// get clusterRoleName from permission type
+	clusterRoleName := permission.ClusterRoleName
+
+	// get groupname from GroupPermission object
+	groupName := instance.Spec.GroupName
+
+	var tmpArray []string
+
+	for _, a := range instance.Spec.ClusterPermissions {
+		clusterRoleName := a.ClusterRoleName
+		clusterRoleBindingName := clusterRoleName + "-" + groupName
+
+		tmpArray = append(tmpArray, clusterRoleBindingName)
+	}
+
+	// clusterRoleBindingItems := clusterRoleBindingList.Items
+
+	if checkClusterRoleBindingName(tmpArray, clusterRoleBindingList) {
+		return reconcile.Result{}, nil
+	} else {
+		// create a clusterRoleBinding
+		// this is absolutely NOT gonna work... plz advice... TODO
+		newCRB := newClusterRoleBinding(instance, permission)
+		if newCRB != nil {
+			groupPermissionConditions := instance.Status.Conditions
+
+			// make new condition
+			newCondition := managedv1alpha1.Condition{
+				LastTransitionTime: metav1.Now(),
+				ClusterRoleName:    clusterRoleName,
+				Status:             true,
+				State:              "Created",
+			}
+
+			// append new condition to conditions array
+			groupPermissionConditions = append(groupPermissionConditions, newCondition)
+
+			// update GroupPermission object with new conditions
+			instance.Status.Conditions = groupPermissionConditions
```

Comment:
I'd say merge this line and 175. `instance.Status.Conditions = append(groupPermissionConditions, newCondition)`